### PR TITLE
for #1812 make ShardingTransactionManager support multi instance.

### PR DESCRIPTION
**Pull request by cherrylzhao**

### CODE REVIEW

File: sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ShardingTransactionManagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return doGetConnection(uniqueKey + dataSourceName);
+    }
+    
+    /**
+     * Do get connection.
+     *
+     * @param dataSourceName data source name
+     * @return connection
+     */
+    public abstract Connection doGetConnection(String dataSourceName);
```
Comment (terrymanu) [inline sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ShardingTransactionManagerAdapter.java]: create new class named ResourceDataSource to deal with resource name, maybe is better than use `getConnection` and `doGetConnection`

File: sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ShardingTransactionManagerEngine.java [hunk 2]
```diff
@@ -36,24 +34,35 @@
  *
  * @author zhaojun
  */
-@NoArgsConstructor(access = AccessLevel.PRIVATE)
 @Slf4j
 public final class ShardingTransactionManagerEngine {
     
-    private static final Map<TransactionType, ShardingTransactionManager> TRANSACTION_MANAGERS = new HashMap<>();
+    private final Map<TransactionType, ShardingTransactionManager> cachedTransactionManager = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (terrymanu) [reply sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ShardingTransactionManagerEngine.java]: So the name should not called as cached, is it?

[the remark this answers is not in the document]
Comment (cherrylzhao) [reply sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ShardingTransactionManagerEngine.java]: that's right, let me revise it.

File: sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ShardingTransactionManagerAdapter.java (hunk 1 above)
Comment (cherrylzhao) [reply sharding-transaction/sharding-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ShardingTransactionManagerAdapter.java]: I have fixed it using ResourceDataSource.

**Pull request merged**